src/components/PopUpWithForm.js

```diff
@@ -3,34 +3,59 @@ import PopUp from "./PopUp.js";
 export default class PopUpWithForm extends PopUp {
   constructor(popUpSelector, handleFormSubmit) {
     // Why super(popUpSelector) did not work? Re
+    // PopUp accepts an object --- in its constructor
+    // vs. PopUpwithForm inherits PopUp's functions and instances by super({popUpSelector}) but that does not mean popUpwithForm
+    // necessarily have to accept an object in constructor -> because super({popUpSelector}) is handling it
     super({ popUpSelector });
 
     this._popUpForm = this._popUpElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
+
+    this._submitButton = this._popUpForm.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  setInputValues(data) {
+    this._inputList.forEach((input) => {
+      // Here you insert the `value` by the `name` of the input
+      input.value = data[input.name];
+    });
   }
 
   _getInputValues() {
-    this._inputList = this._popUpForm.querySelectorAll(".modal__input");
-    this._inputValues = {};
+    //All elements found every time you call a method should be defined as class fields in the constructor, in order
+    // not to waste resources on searching them again and again
+    //but don't we get different value whenever there's new userinput?
 
+    this._inputValues = {};
     this._inputList.forEach((input) => {
       this._inputValues[input.name] = input.value;
-      console.log(this._inputValues);
-      console.log(this._inputValues.title);
-      console.log(this._inputValues.url);
     });
 
     return this._inputValues;
   }
 
+  setSaving(isSaving) {
+    console.log(`this._submitButton: ${this._submitButton}`);
+
+    console.log(`isSaving: ${isSaving}`);
+    if (isSaving) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
```

Comment:
**COULD BE IMPROVED**


`"Saving..."` is hardcoded here

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```